pkgs/applications/office/vnote/default.nix

```diff
@@ -0,0 +1,56 @@
+{ stdenv, fetchFromGitHub, qt5, hicolor-icon-theme, makeDesktopItem }:
+
+let
+  description = "A note-taking application that knows programmers and Markdown better";
+  desktopItem = makeDesktopItem {
+    name = "VNote";
+    exec = "vnote";
+    icon = "vnote";
+    comment = description;
+    desktopName = "VNote";
+    categories = "Office";
+  };
+in stdenv.mkDerivation rec {
+  version = "2.5";
+  name = "vnote-v${version}";
+
+  src = fetchFromGitHub {
+    owner = "tamlok";
+    repo = "vnote";
+    fetchSubmodules = true;
+    rev = "8d780bd5898464139e252623ffaccd966e093ff1";
+    sha256 = "17nl4z1k24wfl18f6fxs2chsmxc2526ckn5pddi2ckirbiwqwm60";
+  };
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,share/applications,share/icons/hicolor}
+
+    cp ./src/VNote $out/bin/vnote
+    cp ${desktopItem}/share/applications/* $out/share/applications/
+
+    pushd .
+    cd $out/share/icons/hicolor
+    mkdir -p ./{scalable,16x16,32x32,48x48,64x64,128x128,256x256}/apps
+    popd
+
+    cp src/resources/icons/vnote.svg $out/share/icons/hicolor/scalable/apps/vnote.svg
+    for x in 16x16 32x32 48x48 64x64 128x128 256x256; do
+      cp src/resources/icons/$x/vnote.png $out/share/icons/hicolor/$x/apps/vnote.png
+    done
+
+    runHook postInstall
+  '';
+
+  nativeBuildInputs = [ qt5.qmake ];
```

Comment:
I notice that `vnote` includes a `CMakeLists.txt` in their repo. Is there a particular reason you used `qmake` file instead? Just curious.

Reply:
Tried without `qmake` and it does not seems to be able to figure out how to build it without it. So, will keep it as native build dependency for now.

Reply:
@kuznero Yeah I just took a look and I'm under the impression you might need to manually specify a bunch of qt paths if you use `cmake`... forget that option.